Namitha:  Phase 4 -Announcements Frontend

Namitha7070 opened the pull request:
# Description
 (PRIORITY MED) Harshitha : Phase 4 - Announcements Frontend

CSS module needs to be added to the page (and ALL pages) too
Result / Features:
Educators can post and edit announcements via modal.
Students see announcements in a clean board format.
New announcements appear in real time without reload.
Supports filtering by “audience.”
UI Components:
AnnouncementsBoard.jsx – list of announcements on dashboard.
AnnouncementModal.jsx – create/edit announcement form.

Trigger Points:
Educator → post from dashboard.
Student → view-only.

Error Handling UI:
Form validation on title/body.

## Related PRS (if any):
This frontend PR is related to the #XXX backend PR.
To test this backend PR you need to checkout the #XXX frontend PR.
…

## Main changes explained:
Complete announcements system with educator/student roles
- Create/Edit announcements modal with validation
- Comprehensive filtering (audience, search, course, date)
- Role-based access control using authentication system
- Professional UI matching design requirements
- Responsive design with mobile support
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user(for Educator's View), volunteer(for Student View)
5. go to dashboard→ http://localhost:5173/educationportal/announcements
6.  verify if this new feature works by creating a new announcement and try different filters(Student & Educator)

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/7ce31b3b-8259-450b-a8b8-d8ef264c0fa5

Namitha7070 committed:
feat: Implement Phase 4 Announcements Frontend

- Add AnnouncementsPage with full layout and filtering system
- Create AnnouncementModal for creating/editing announcements
- Implement AnnouncementsBoard for displaying announcements
- Add comprehensive filtering: audience, search, course, date filters
- Include role-based access control (educators vs students)
- Support dark mode throughout all components
- Add form validation and unsaved changes detection
- Integrate with Redux for auth and theme state
- Add protected route at /educationportal/announcements

Namitha7070 committed:
Merge branch 'development' into feature/phase-4-announcements-frontend

Anusha-Gali reviewed (changes requested):
Hi Namitha,

I have reviewed your PR locally, i am able to create different types of announcement but issue occur in :
(In Admin/Educator login)
- When i try to edit the educators only to student or vice versa, the page breaks and upon refresh i loose all my activity.
- The dark mode is not enabled for the page as well as the cancel on creating announcement

https://github.com/user-attachments/assets/b4070af8-8d4a-4f3a-a230-4ea15b3a24bb

And after creating an announcement in educator, i logged into volunteer account but am not able to see the latest announcement.
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 30 05 PM" src="https://github.com/user-attachments/assets/14f87cf0-a24f-4b25-b44f-99e33390ee76" />
<img width="1272" height="762" alt="Screenshot 2026-01-22 at 11 30 08 PM" src="https://github.com/user-attachments/assets/81794e51-62ee-4517-a1b5-9d88adea0983" />

Let me know if any of the changes requested isn't in scope.

rohanrastogi311 reviewed (approved):
Hi Namitha, 

Well done on this implementation. Working well. 

<img width="1726" height="1039" alt="PR 4296 Screenshot" src="https://github.com/user-attachments/assets/62cfb546-07cf-466f-a087-0bd968618584" />
<img width="1690" height="1031" alt="PR 4296 Screenshot 2" src="https://github.com/user-attachments/assets/e075b5b3-8a81-4533-a0a3-012ae6821acc" />
<img width="1725" height="1040" alt="PR 4296 Screenshot 3" src="https://github.com/user-attachments/assets/3f28455b-065c-4416-a0bd-63e90bb39542" />

HemanthNidamanuru reviewed (changes requested):
Hi Namitha,

Thanks for working on this. A few issues came up during testing:

1. The page isn't dark mode optimized. The calendars and announcement cards aren't aligned with the dark theme.
2. When selecting "Everyone" in the audience, the Student tag is still getting attached, so filtering doesn't work correctly.
3. Editing an announcement isn't possible, as Anusha also pointed out.

<img width="1905" height="987" alt="Image" src="https://github.com/user-attachments/assets/2b2d17eb-8139-4982-9561-ce1abe35b032" />
<img width="1882" height="1042" alt="Image" src="https://github.com/user-attachments/assets/8631eda2-4a1e-402b-8cb5-07b72c44b4d8" />
<video src="https://github.com/user-attachments/assets/ade0f120-50ec-4541-9dd9-128041be9187" controls width="100%"></video>

Namitha7070 committed:
fix: align announcements dark mode with app navy blue theme, fix audience filter and edit persistence

Namitha7070 committed:
fix(lint): fix prettier and eslint errors in Announcements components

HemanthNidamanuru reviewed (dismissed):
Hi Namitha,

Thank you for working on the improvements. All the issues have been resolved and everything looks good.

<img width="1898" height="1039" alt="Image" src="https://github.com/user-attachments/assets/8e8a1d7b-740a-4f86-8002-3f033915476f" />
<img width="1899" height="1044" alt="Image" src="https://github.com/user-attachments/assets/9efbd557-cada-43ed-81cc-d36e759a0213" />
<img width="1919" height="1037" alt="Image" src="https://github.com/user-attachments/assets/6d2e0f24-a2a0-4b99-aa0f-028ecd0492ee" />
<img width="1900" height="1036" alt="Image" src="https://github.com/user-attachments/assets/9f9c3a99-3d3d-4966-a620-c9914d59e11f" />

Namitha7070 committed:
fix: resolve 35 SonarQube quality gate issues in Announcements components

- Add PropTypes to AnnouncementModal and AnnouncementsBoard components
- Extract nested ternary operations into helper functions (getBadgeClass, getSubmitButtonText, getEmptyMessage, getAudienceLabel, getButtonStyle)
- Extract embedded view rendering to reduce Cognitive Complexity
- Remove unused setLoading variable in AnnouncementsBoard
- Fix optional chain expression for announcement.course
- Fix negated condition (length !== 1 -> length === 1)
- Fix exception handling: log parse error instead of silently ignoring
- Extract getUserRole and other helpers to module level in AnnouncementsPage
- Fix CSS contrast issues in AnnouncementModal.module.css (modalHeader, saveButton)
- Fix CSS contrast issues in AnnouncementsBoard.module.css (editButton hover)
- Remove duplicate CSS selector in AnnouncementModal.module.css

Namitha7070 committed:
fix: remove duplicate background-color before background shorthand in modalHeader

Namitha7070 committed:
fix: resolve remaining SonarQube issues - Set for EDUCATOR_ROLES, extract renderGridContent and renderSidebar to eliminate nested ternaries and reduce cognitive complexity

Namitha7070 committed:
fix: reduce AnnouncementsPage cognitive complexity by extracting inputStyle and map helper to module level

Namitha7070 committed:
fix: extract FilterSidebar and ActiveFiltersBar to module level to reduce AnnouncementsPage cognitive complexity from 18 to 14

Namitha7070 committed:
fix: reduce AnnouncementsPage cognitive complexity to 13 and fix AnnouncementModal prettier formatting

Namitha7070 committed:
fix: reduce AnnouncementsPage cognitive complexity to 12 and fix emoji icons

Namitha7070 committed:
refactor: move AnnouncementsPage handlers to module-level factories to lower complexity
